packages/baseweb-vscode-extension/ext-src/coloring.ts

```diff
@@ -0,0 +1,100 @@
+import * as vscode from 'vscode';
+import {LightTheme, DarkTheme} from 'baseui';
+
+export default (context: vscode.ExtensionContext) => {
+  let timeout: NodeJS.Timer | undefined = undefined;
+  let activeEditor = vscode.window.activeTextEditor;
+
+  // create a decorator type that we use to decorate small numbers
+  const getColorDecorationType = (coloringStyle: string, colorVal: string) =>
+    vscode.window.createTextEditorDecorationType({
+      borderWidth:
+        coloringStyle === 'border'
+          ? '2px'
+          : coloringStyle === 'underline'
+          ? '0 0 2px'
+          : '0',
+      borderStyle: 'solid',
+      borderColor: colorVal,
+      backgroundColor:
+        coloringStyle === 'background' ? colorVal : 'transparent',
+      opacity: 0.5,
+    });
+
+  function updateDecorations() {
+    if (!activeEditor) {
+      return;
+    }
+
+    const workspaceConfig = vscode.workspace.getConfiguration('baseweb');
+    const isColoringOn = workspaceConfig.get('theme.coloring.enabled');
+    if (!isColoringOn) {
+      return;
+    }
+    const themeMode = workspaceConfig.get('theme.coloring.source');
+    const theme = themeMode === 'Light' ? LightTheme : DarkTheme;
+    const coloringStyle = workspaceConfig.get('theme.coloring.style');
+
+    const regEx = /colors\.\w+/g;
```

Comment:
this regexp is probably too permissive? (problem with coloring neighbor characters)

Reply:
Improved the regexp ^